### PR TITLE
fix : Improve map fragment

### LINKED ISSUE #1743: Improve Map fragment
**Actual Behaviour**

Currently map fragment is having some space before auto complete text view in actionbar. And marker info is not being showed on clicking item in auto complete text view suggestion. And also serch behaviour in Map fragment is different from other fragments.

**Expected Behaviour**

- There should not be any space before auto complete textview.
- Use Search view with SearchAutoComplete instead of auto complete text view to make serch feature uniform throught out the app.
- It should display marker info(location name) on clicking item from suggestion.

**Screenshots of the issue**

map

**Would you like to work on the issue?**

Yes

**Target Milestone to solve the issue**

[Milestone 8](https://github.com/fossasia/open-event-android/milestone/31)

### PR DESCRIPTION
**Pull request by Shailesh351**
Fixes #1743 

Changes: 
- Fixed padding before autocomplete search view
- Changed autocomplete textview to SearchView.SearchAutoComplete to make UI consistent
- Showed marker info on clicking on suggestion
- Added null check for event and location
- Removed sharing option from fDroid version
- Clicking map icon from sesssion detail activity and location activity directly focus on corresponding location and shows marker info.

Screenshots for the change: 

videotogif_2017 06 29_18 48 32

### CODE REVIEW

**Comment by Shailesh351** [comment]
@arp95 @shreyasnbhat @iamareebjamal @rohanagarwal94 Please review.

**Comment by Shailesh351** [comment]
Which one will be better  ?

videotogif_2017 06 29_19 08 35

**Comment by shreyasnbhat** [comment]
@Shailesh351 I think that the 2nd one would be better

**Comment by arp95** [comment]
i think the same. 2nd will be better.

**Comment by Shailesh351** [comment]
Do not merge this PR. Some work is remaining. Working on it

**Comment by Shailesh351** [comment]
Guys now clicking map icon from sesssion detail activity and location activity directly focus on corresponding location and shows marker info of that location.

Ready to merge.
@iamareebjamal @arp95 @shreyasnbhat @rohanagarwal94 Please review now.

**Comment by Shailesh351** [comment]
videotogif_2017 07 02_01 25 20

File: android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java [hunk 1]
```diff
@@ -223,22 +217,6 @@ public void onDestroy() {
...
-                intent.setType("text/plain");
-                startActivity(Intent.createChooser(intent, "Share URL"));
-                break;
-            default:
-                //do nothing
-        }
-        return super.onOptionsItemSelected(item);
-    }
-
 //    private void launchDirections() {
```
Comment (iamareebjamal) [inline android/app/src/fdroid/java/org/fossasia/openevent/fragments/OSMapFragment.java]: Can you remove the comments too? Thanks

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**